sql/moz-fx-data-shared-prod/mozilla_vpn_derived/funnel_product_page_to_subscribed_v1/query.sql

```diff
@@ -107,15 +113,51 @@ flows AS (
     LOGICAL_OR(event_type = "fxa_subscribe_coupon - fail") AS subscribe_coupon_fail,
     LOGICAL_OR(event_type = "fxa_subscribe_coupon - success") AS subscribe_coupon_success,
   FROM
-    mozdata.firefox_accounts.fxa_content_auth_stdout_events
+    events
   WHERE
-    IF(@date IS NULL, DATE(`timestamp`) < CURRENT_DATE, DATE(`timestamp`) = @date)
+    IF(
+      @date IS NULL,
+      -- The fully FxA-based funnel for VPN was enabled on 2021-08-25.
+      (partition_date >= "2021-08-25" AND partition_date < CURRENT_DATE),
+      partition_date = @date
+    )
+    AND flow_id IS NOT NULL
   GROUP BY
     partition_date,
     flow_id
   HAVING
     -- NOTE: flows near date boundaries may not meet this condition for all dates
-    LOGICAL_OR(service = "guardian-vpn")
+    LOGICAL_OR(
+      service = "guardian-vpn"
+      OR events.entrypoint = "www.mozilla.org-vpn-product-page"
+      OR events.utm_source = "www.mozilla.org-vpn-product-page"
+      OR (
+        event_type = "fxa_rp_button - view"
+        AND (
+          -- The www.mozilla.org navbar CTA button was changed to link to VPN for Firefox users
+          -- on 2021-09-02, then attribution was implemented for it on 2021-09-15.
+          (
+            service IS NULL
+            AND events.utm_source = "www.mozilla.org"
+            AND events.utm_campaign = "navigation"
+            AND (partition_date BETWEEN "2021-09-15" AND "2021-12-08")
```

Comment:
I put parentheses around `BETWEEN` expressions to prevent `bqetl format` from formatting them confusingly.